plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py

```diff
@@ -187,6 +217,41 @@ def setup(
         )
         self.sweep_dir = config.hydra.sweep.dir
 
+    def _get_directions(self) -> List[str]:
+        if isinstance(self.direction, MutableSequence):
+            return [d.name if isinstance(d, Direction) else d for d in self.direction]
+        elif isinstance(self.direction, str):
+            return [self.direction]
+        return [self.direction.name]
+
+    def _configure_trials(
+        self,
+        trials: List[Trial],
+        search_space: Dict[str, BaseDistribution],
+        fixed_params: Dict[str, Any],
+    ) -> Sequence[Sequence[str]]:
+        overrides = []
+        for trial in trials:
+            for param_name, distribution in search_space.items():
+                assert type(param_name) is str
+                trial._suggest(param_name, distribution)
+            for param_name, value in fixed_params.items():
+                trial.set_user_attr(param_name, value)
+
+            if self.custom_search_space:
+                self.custom_search_space(self.config, trial)
+
+            overlap = trial.params.keys() & trial.user_attrs
+            assert len(overlap) == 0, (
+                "Overlapping fixed parameters and search space parameters found!"
+                f"Overlapping parameters: {list(overlap)}"
+            )
```

Comment:
Can this error be triggered by the user making a mistake with their config? If so, this should be a `ValueError` instead of an `AssertionError`.

Reply:
Yes it can, I will change this